### PR TITLE
Re-enable gossip ref tests

### PR DESCRIPTION
**Pull request by gfukushima**
## PR Description
This PR re-enables a few gossip ref-tests and fix some edge cases around chain initialisation for ref tests that start with a block that is past the initial state. Also handle fork initialisation using the ref test file `config.yaml`.

## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [ ] I thought about adding a changelog entry, and added one if I deemed necessary.

---

> [!NOTE]
> **Medium Risk**
> AttestationValidator gossip outcomes change for attestations when fork-choice ancestry is incomplete (IGNORE instead of REJECT), which affects how attestations are accepted on the network path.
> 
> **Overview**
> Re-enables **networking gossip** reference tests for beacon attestations, aggregates, and BLS-to-execution changes (replacing suite-wide ignores) and tightens how fixtures bootstrap chain state and assert outcomes.
> 
> **Test harness:** `createAnchorFromState` rolls the fixture state back to `latest_block_header.slot` when the SSZ state is ahead of the anchor block, so the store/proto-array anchor slot matches the header and `getAncestor` works in gossip setups. Attestation and aggregate executors expect **ignore** when fixtures use a custom `finalized_checkpoint` Teku does not model (no finalized-ancestor check in production). BLS-to-execution tests read `config.yaml` for `CAPELLA_FORK_EPOCH` and assert **ignore** before Capella.
> 
> **Production:** `AttestationValidator` splits LMD target ancestry: missing ancestor in fork choice → **IGNORE** (insufficient data); resolved ancestor ≠ target root → **REJECT**.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 9c5f04981737ca8aecf219a4fd761e7687b39228. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by zilm13** [comment]
Ohh, great, I've started it in #10724 but stuck a bit. Yours looks better, reviewing

File: eth-reference-tests/src/referenceTest/java/tech/pegasys/teku/reference/phase0/gossip/GossipBeaconAttestationTestExecutor.java [hunk 1]
```diff
@@ -178,6 +178,20 @@ public void runTest(final TestDefinition testDefinition) throws Throwable {
         continue;
       }
 
+      // Custom finalized_checkpoint: the fixture pins a finalized root not in our chain, so by
+      // definition the attestation's block cannot descend from it. Per spec, this is IGNORE. Teku's
+      // validator skips this check (it relies on the proto-array invariant that every node
+      // descends from the finalized block — an invariant the fixture deliberately violates), so
+      // handle it here.
+      if (hasCustomFinalizedCheckpoint) {
```
Comment (zilm13) [inline eth-reference-tests/src/referenceTest/java/tech/pegasys/teku/reference/phase0/gossip/GossipBeaconAttestationTestExecutor.java:189]: So we just ignore these cases and don't run the checks for them? 
I don't think it's a good thing to hide something like that in an executor. If we are not supporting this type of tests, they should be in ignored list or smth like this clearly highlighting that we don't run these tests. Whenever we skip the test it could avoid revealing the bug in our code, so it shouldn't be hidden.

File: eth-reference-tests/src/referenceTest/java/tech/pegasys/teku/reference/phase0/gossip/GossipBeaconAggregateAndProofTestExecutor.java [hunk 2]
```diff
@@ -164,6 +164,20 @@ public void runTest(final TestDefinition testDefinition) throws Throwable {
         continue;
       }
 
+      // Custom finalized_checkpoint: the fixture pins a finalized root not in our chain, so by
+      // definition the aggregate's block cannot descend from it. Per spec, this is IGNORE. Teku's
+      // validator skips this check (it relies on the proto-array invariant that every node
+      // descends from the finalized block — an invariant the fixture deliberately violates), so
+      // handle it here.
+      if (hasCustomFinalizedCheckpoint) {
```
Comment (zilm13) [inline eth-reference-tests/src/referenceTest/java/tech/pegasys/teku/reference/phase0/gossip/GossipBeaconAggregateAndProofTestExecutor.java:175]: same comment as in `GossipBeaconAttestationTestExecutor`

File: eth-reference-tests/src/referenceTest/java/tech/pegasys/teku/reference/phase0/gossip/GossipBlsToExecutionChangeTestExecutor.java [hunk 3]
```diff
@@ -130,6 +165,24 @@ public void runTest(final TestDefinition testDefinition) throws Throwable {
     }
   }
 
+  private static UInt64 readCapellaForkEpoch(final TestDefinition testDefinition) {
```
Comment (zilm13) [inline eth-reference-tests/src/referenceTest/java/tech/pegasys/teku/reference/phase0/gossip/GossipBlsToExecutionChangeTestExecutor.java]: this could be done in more elegant way, see https://github.com/Consensys/teku/pull/10724/changes/a496358405b45f0790380b7874b611b2ac3f0f98#diff-6dc1af84fae1e2eec47d548f5310a306de87fc7c39bcd5c6ef31cd44a00dd81a

File: eth-reference-tests/src/referenceTest/java/tech/pegasys/teku/reference/phase0/gossip/GossipBlsToExecutionChangeTestExecutor.java [hunk 4]
```diff
@@ -72,11 +81,37 @@ public void runTest(final TestDefinition testDefinition) throws Throwable {
...
+        final UInt64 currentSlot = spec.getCurrentSlotFromTimeMillis(messageTimeMs, genesisTimeMs);
+        final UInt64 currentEpoch = spec.computeEpochAtSlot(currentSlot);
+        if (currentEpoch.isLessThan(capellaForkEpochOverride)) {
+          assertThat(message.getExpected())
+              .describedAs(
+                  "Expected ignore for BLS-to-execution-change %s — current epoch %s is pre-capella"
+                      + " (CAPELLA_FORK_EPOCH override = %s)",
+                  message.getMessage(), currentEpoch, capellaForkEpochOverride)
+              .isEqualTo("ignore");
+          continue;
```
Comment (zilm13) [inline eth-reference-tests/src/referenceTest/java/tech/pegasys/teku/reference/phase0/gossip/GossipBlsToExecutionChangeTestExecutor.java]: So we skip running these tests
I think we shouldn't have any hidden ignores, they should be explicitely listed with reasoning.

File: eth-reference-tests/src/referenceTest/java/tech/pegasys/teku/reference/phase0/gossip/GossipBeaconAttestationTestExecutor.java (hunk 1 above)
Comment (gfukushima) [reply eth-reference-tests/src/referenceTest/java/tech/pegasys/teku/reference/phase0/gossip/GossipBeaconAttestationTestExecutor.java:189]: these are a subset of the suite, I think it worth adding a skip on those, i'll improve the comment to show where why we're skipping those, we do have a reason in a comment here: https://github.com/Consensys/teku/blob/38f7b080ce1ea5d0074b7fcbf1cb52a71c1f56da/ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/validation/AttestationValidator.java#L226

File: eth-reference-tests/src/referenceTest/java/tech/pegasys/teku/reference/phase0/gossip/GossipBlsToExecutionChangeTestExecutor.java (hunk 3 above)
Comment (gfukushima) [reply eth-reference-tests/src/referenceTest/java/tech/pegasys/teku/reference/phase0/gossip/GossipBlsToExecutionChangeTestExecutor.java]: adopted your approach, thanks!

**Comment by gfukushima** [comment]
closing since it's getting done in #10724
